File: Producto/VeciZervi-BackEnd/backend/src/main/java/com/vecizervi/backend/model/Usuario.java
package com.vecizervi.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "usuarios")
public class Usuario {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usuario") 
    private Long id;

    @Column(unique = true, nullable = false)
    private String rut;

    private String nombres;
    private String apellidos;
    private LocalDate fechaNacimiento;

    @Column(unique = true, nullable = false)
    private String correo;

    @Column(name = "password", nullable = false)
    private String contrasenaEnCriptada; 

    private Integer intentosFallidos = 0; 
    private LocalDateTime cuentaBloqueadaHasta; 
    private String tokenRecuperacion;
}

File: Producto/Android Studio/VeciZerviGub/app/src/main/java/com/example/vecizervi/data/models/Usuario.kt
package com.example.vecizervi.data.models

import com.google.gson.annotations.SerializedName

data class Usuario(
    @SerializedName("idUsuario") val idUsuario: Int? = null,
    @SerializedName("rut") val rut: String = "",
    @SerializedName("nombres") val nombres: String = "",
    @SerializedName("apellidos") val apellidos: String = "",
    @SerializedName("fechaNacimiento") val fechaNacimiento: String = "",
    @SerializedName("correo") val correo: String = "",
    @SerializedName("password") val password: String = "",
    @SerializedName("herramientas_propias") val herramientasPropias: String? = null,
    @SerializedName("calificacion_promedio") val calificacionPromedio: Double = 0.0,
    @SerializedName("rol") val rol: String = "USER"
)
